### DISCUSSION TITLE
bug: icons not visible under dark color schemes

### DISCUSSION THREAD

**Discussion by AyushRawal** [eza-community/eza#442] [Development]
![image](https://github.com/eza-community/eza/assets/32910415/848b6577-4933-43d7-8674-9e529bc17b04)

They are certainly present, just not visible:
![image](https://github.com/eza-community/eza/assets/32910415/a8507826-4dec-4e7f-9c95-94bb731221df)

I am using the latest version:

```
$ eza --version
eza - A modern, maintained replacement for ls
v0.13.0 [+git]
https://github.com/eza-community/eza
```

**Comment by PThorpe92** [comment]
what version of NerdFonts are you using?

**Comment by cfxegbert** [comment]
The icon should be the same color as the filename.  What command line options are you using?

Edit: Looking at the code it could be using the background color.

**Comment by sbatial** [comment]
And what color scheme?
I couldn't reproduce it on a dark scheme (i'd have to look which one I'm currently using but it should be carbonfox)

![image](https://github.com/eza-community/eza/assets/46973479/7187e574-4422-4b64-b19e-c9c3d4d7675a)

**Comment by AyushRawal** [comment]
well this sucks, i am using [vivid](https://github.com/sharkdp/vivid) to generate ls-colors
`export LS_COLORS=$(vivid generate gruvbox-dark-hard)`

resetting ls-colors fixes this issue

also, i am using nerd fonts v2.3.3 since v3.0.0 introduced some breaking changes, but i also tried with v3.0.2 (latest)

i am using gruvbox dark color scheme (hard variant)

**Comment by ghost** [comment]
Yeah, that theme seems to use white on black background for dot files, and eza apparently uses the filename's foreground color for the icon unless the filename has a background color and then it uses the background color:

$ LS_COLORS="$(vivid generate gruvbox-dark-hard)" EXA_COLORS="" eza -l --icons -a ~ 

![image](https://github.com/eza-community/eza/assets/56895621/1aedc640-a47f-4e9b-9446-684782722caa)

as opposed to 

$ LS_COLORS="$(vivid generate lava)" EXA_COLORS="" eza -l --icons -a ~

![image](https://github.com/eza-community/eza/assets/56895621/09c12291-7b58-4800-b975-40b072bb3194)

and in your terminal's theme the black background matches the terminal background.

**Comment by AyushRawal** [comment]
For now, I have modified my colorscheme to remove background color.
```diff
core:
  normal_text:
    foreground: fg
-   background: bg

  regular_file:
    foreground: fg
-   background: bg

  reset_to_normal: {}

```
Should we close this issue ?

**Comment by sbatial** [comment]
> unless the filename has a background color and then it uses the background color

I finally understood that sentence (probably because that behaviour is indeed quite odd).

So yea... that should not be desired imo

> Should we close this issue ?

Well I think it is still a problem that would need a fix or at least a justification. So I'd leave it open.
The icons should just use the foreground color of the filename in my opinion.

I'll try to look into it should I have the time.

**Comment by ghost** [comment]
It can run into trouble either way. If the filename has different foreground and background colors then either can match the terminal background and it'll still be visible, but making the icon foreground match either with no background can lead to its disappearing. So maybe the icon should use both the same foreground and background colors.

**Comment by sbatial** [comment]
> So maybe the icon should use both the same foreground and background colors.

That's what I meant. :+1: 

Because I think it should be reasonable to assume that the theme is chosen/created in such a way that filenames are properly legible so I don't see a reason why the icons shouldn't just use the texts properties.

**Comment by sbatial** [comment]
So I found the change in question and it seemed to be a deliberate decision.

So at that point I think it is best to convert this into a discussion to talk about what the best way forward would be. (Because I strongly believe in only throwing stuff away when you know why it was put there in the first place)

So far my findings/experiments are as far as follows:

- 976db01b3e93aa537e1a682e61a8e02febfa6cad introduced the change (as you can see the message says "Prefer background colour when painting icons" so there was some reason to do it)
- If we decide to keep the current situation the icons should the have their colors fully inverted (so that they are still visible) which would look something like this:
![image](https://github.com/eza-community/eza/assets/46973479/0a29f029-81cf-456e-8421-3a3ee681ea7c)
^ This is a first attempt. I would say that the text theming shouldn't be affected in the end as it is here
![image](https://github.com/eza-community/eza/assets/46973479/690c087a-236b-4ab5-9568-d51d8fd9d596)
(those are using the gruvbox-(dark|light)-hard themes)

**Reply by sbatial** [reply]
Addition: I think the reasoning behind the "use background color if it exists" presumes that if the theme wants to use the default terminal background it shouldn't set it manually but just use no background and so if a background is manually set it has to be distinct from the usual terminal background.

(I feel like this this does not _not_ make sense but is a kind of brittle assumption.)

Below: eza using the _gruvbox-light-hard_ scheme on a terminal using dayfox (if I'm not remembering wrong)
As one can see there are some files which have a slightly different background color and also basically invisible icons.

![image](https://github.com/eza-community/eza/assets/46973479/b66d217e-f6b7-41ac-ac7a-4d2444367a9d)

**Comment by sbatial** [comment]
(@cafkafk I felt the _Development_ category looked most appropriate, looking at the other discussions. If there is a better suited one let me know. thanks)

**Reply by cafkafk** [reply]
No this is perfect IMO, and great decision to move it here.
